### PR TITLE
ppo updates

### PR DESCRIPTION
**Pull request by budzianowski**
- remove second critic computation
- add logic for multiple updates and batching
- add value function clipping
- add first ppo tests - gae

### CODE REVIEW

File: ksim/task/ppo.py [hunk 1]
```diff
@@ -235,21 +236,32 @@ def compute_loss(
...
+
+        value_loss = jax.lax.cond(
+            self.config.use_clipped_value_loss,
+            lambda: jnp.mean(
+                jnp.maximum(
+                    (value_pred - returns) ** 2,
+                    (
+                        returns
+                        + jnp.clip(value_pred - returns, -self.config.clip_param, self.config.clip_param)
+                        - returns
```
Comment (michael-lutz) [inline ksim/task/ppo.py]: I might be wrong, but I believe this just resolves to the clipping term?

Comment (michael-lutz) [reply ksim/task/ppo.py]: For clarity, let's maybe use a helper function? I think the following should still be jittable

```
def clipped_value_loss():
    error = value_pred - returns
    clipped_error = jnp.clip(error, -self.config.clip_param, self.config.clip_param)
    return 0.5 * jnp.mean(jnp.maximum(error ** 2, clipped_error ** 2))
```

Also, I wonder if we should use a separate clipping hyperparam for the policy & value loss? I don't have the best intuition on the loss scaling so I'll defer to you!

File: ksim/task/ppo.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 from ksim.task.rl import RLConfig, RLTask
 from ksim.types import ModelObs, ModelOut
 
+_EPS = 1e-8
```
Comment (michael-lutz) [inline ksim/task/ppo.py:22]: Nice!

File: ksim/task/ppo.py (hunk 1 above)
Comment (budzianowski) [reply ksim/task/ppo.py]: I always saw them being the same for both policy and value loss.

**Comment by budzianowski** [comment]
@michael-lutz - also currently values and values_pred are the same which is wrong. The former should be the target value so this should be fixed here as well (by storing in the batch)
